### PR TITLE
Panel does work when repeater is paged

### PR DESCRIPTION
**Pull request by Mylan719**
Paging using static command does not work with the loadable panel.

The newly loaded items will never get loadable panels to load.

### CODE REVIEW

**Comment by Mylan719** [comment]
This is just a repro sample the bug remains to be found.

File: Controls/LoadablePanel/src/DotVVM.Contrib.Samples/Controls/Pager/Pager.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using DotVVM.Framework.Controls;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DotVVM.Contrib.Samples.Controls.Pager
+{
+    public static class PagerExtensions
+    {
+        public static void GoToPage(IPageableGridViewDataSet set, int index, Func<IPageableGridViewDataSet> loader)
+        {
+            throw new NotImplementedException("JS");
```
Comment (MichalTichy) [inline Controls/LoadablePanel/src/DotVVM.Contrib.Samples/Controls/Pager/Pager.cs]: Please add better description of why this method is not implemented.

File: Controls/LoadablePanel/src/DotVVM.Contrib.Samples/Controls/Pager/Pager.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        public static IServiceCollection AddPagerExtensions(this IServiceCollection services)
+        {
+            services.Configure((Action<JavascriptTranslatorConfiguration>)(c =>
+            {
+                RegisterAlertMethod(c, nameof(PagerExtensions.GoToPage), "goToPage");
+            }));
+            return services;
+        }
+
+        private static void RegisterAlertMethod(JavascriptTranslatorConfiguration c, string csharpName, string jsName)
```
Comment (MichalTichy) [inline Controls/LoadablePanel/src/DotVVM.Contrib.Samples/Controls/Pager/Pager.cs]: I don't get the name of this method, can you elaborate?

**Review by MichalTichy — CHANGES_REQUESTED** [review]

File: Controls/LoadablePanel/src/DotVVM.Contrib.Samples/Controls/Pager/Pager.cs (hunk 2 above)
Comment (Mylan719) [reply Controls/LoadablePanel/src/DotVVM.Contrib.Samples/Controls/Pager/Pager.cs]: Oh, forgot to rename that one...

File: Controls/LoadablePanel/src/DotVVM.Contrib/LoadablePanel.cs [hunk 3]
```diff
@@ -72,15 +72,28 @@ public bool HideUntilLoaded
             DotvvmProperty.Register<bool, LoadablePanel>(t => t.HideUntilLoaded);
 
         /// <summary>
-        /// Gets or sets a collection of values of all loadable panels which are currently loading.
+        /// Gets or sets a collection of element ids of all loadable panels which are currently in the loading state.
         /// </summary>
-        public IEnumerable LoadingItems
+        public IEnumerable LoadingElementIds
```
Comment (quigamdev) [inline Controls/LoadablePanel/src/DotVVM.Contrib/LoadablePanel.cs:77]: @Mylan719 I'm not sure whether we use "Ids" anywhere. Please rename LoadingElementIds to LoadingKeySource.

File: Controls/LoadablePanel/src/DotVVM.Contrib.Samples/ViewModels/PagingRepeaterViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                PageSize = 6,
+                TotalItemsCount = 32
+            }
+        };
+
+        public List<string> LoadingItems { get; set; } = new List<string>();
+
+
+        [AllowStaticCommand]
+        public async Task<string> LoadItem(string id)
```
Comment (quigamdev) [inline Controls/LoadablePanel/src/DotVVM.Contrib.Samples/ViewModels/PagingRepeaterViewModel.cs]: This is really bad practice. You have 2 options. 
A) Use static methods in ViewModel with AllowStatiCommand. 
B) Use instantiation methods in static command service. (just separate class, it does not have any ancestor) 

But please do not mix this. We are preparing a Roslyn analyzer to detect this pattern and prevent people from using it.

**Review by quigamdev — CHANGES_REQUESTED** [review]

File: Controls/LoadablePanel/src/DotVVM.Contrib/LoadablePanel.cs [hunk 5]
```diff
@@ -72,15 +72,28 @@ public bool HideUntilLoaded
...
-            DotvvmProperty.Register<IEnumerable, LoadablePanel>(t => t.LoadingItems, null);
+        public static readonly DotvvmProperty LoadingElementIdsProperty =
+            DotvvmProperty.Register<IEnumerable, LoadablePanel>(t => t.LoadingElementIds, null);
+
+        /// <summary>
+        /// Gets or sets a key that identtifies the data displayed in the loadable panel. When the key value changes the panel is reloaded.
+        /// This property needs to be bound if the loadable panel is in a GridView Repeater or other client-updated controls.
+        /// </summary>
+        [MarkupOptions(AllowBinding = true, AllowHardCodedValue = false)]
+        public string Key
```
Comment (tomasherceg) [inline Controls/LoadablePanel/src/DotVVM.Contrib/LoadablePanel.cs]: I'd prefer using some more descriptive name than just `Key` - key to what? 
What about `ContentKey`, `DataKey` or something like that?

Comment (Mylan719) [reply Controls/LoadablePanel/src/DotVVM.Contrib/LoadablePanel.cs]: I renamed the property to `ContentReloadBinding` this should be more acurate

File: Controls/LoadablePanel/src/DotVVM.Contrib.Samples/ViewModels/PagingRepeaterViewModel.cs (hunk 4 above)
Comment (Mylan719) [reply Controls/LoadablePanel/src/DotVVM.Contrib.Samples/ViewModels/PagingRepeaterViewModel.cs]: I reworked the sample project structure to comply with latest DotVVM best practices.

**Review by quigamdev — APPROVED** [review]

**Pull request merged**